imports/client/components/PuzzlePage.tsx

```diff
@@ -1785,6 +1894,11 @@ const PuzzlePage = React.memo(() => {
       Puzzles.findOneAllowingDeleted(puzzleId)
   ), [puzzleDataLoading, puzzleId]);
 
+  const selfUserId = useTracker(() => Meteor.userId()!, []);
+  const users = useTracker(() => {
+    return MeteorUsers.find({}).fetch();
```

Comment:
```suggestion
    return MeteorUsers.find({ hunts: huntId }).fetch();
```

Reply:
Good catch; we definitely only want to complete users associated with this hunt.

Maybe we should enforce that in `ChatNotificationHooks` as well?  Verify that a user is in the same hunt as the chat message before allowing them to be the target of an @-mention.

Reply:
I think so. Also I think in matchUsers, just in case a future subscription pulls in more users.

Reply:
Addressed in 01e2af8ff1ca5bd4f746ef44ecfe224112b60c0a

Reply:
Filtered down to users in the same hunt with a `displayName` set.